### PR TITLE
Remove pods from source control

### PR DESCRIPTION
**Pull request by friederbluemle**
This PR addresses the following issues:

- Updates `.gitignore` to match the default one from React Native
- Adds the missing dependency lock files (they [should be added](https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/))
- Removes the entire `ios/Pods` directory

A note on `ios/Pods`: The folder was already added to `.gitignore` in 62d9576d12e052da3fefb1f03a286214332f1700), however many Pod files were committed to source control in ef621a555d55dc30351b484c08b378d72f932fe9 and dc9c2d8c4cb06f37f6f08ff1e61443c6be2fc9eb, which unfortunately increased the repository size by >15MB (permanently).

### CODE REVIEW

**Comment by friederbluemle** [comment]
@Inbal-Tish - Please review when you have a minute! Thanks 🙏

**Comment by friederbluemle** [comment]
A commit authored `Dec 15` with identical content to the original `Remove pods` commit was pushed to `master` (670be3de9d98ac311f2b5a36059b39ddafb0b0a3). The rebased remainder of this PR now only contains a few minor cleanups to `.gitignore` and adds a `yarn.lock` file.

File: .gitignore [hunk 1]
```diff
@@ -1,11 +1,3 @@
-node_modules
```
Comment (Inbal-Tish) [inline .gitignore]: Why are you removing all these from the ignore file?

File: .gitignore [hunk 2]
```diff
@@ -28,7 +20,6 @@ DerivedData
 *.hmap
 *.ipa
 *.xcuserstate
-# project.xcworkspace
```
Comment (Inbal-Tish) [inline .gitignore]: this one should be added

File: example/.gitignore [hunk 3]
```diff
@@ -20,7 +20,6 @@ DerivedData
 *.hmap
 *.ipa
 *.xcuserstate
-project.xcworkspace
```
Comment (Inbal-Tish) [inline example/.gitignore]: why remove this?

Comment (Inbal-Tish) [reply example/.gitignore]: `yarn-lock` should be added

File: .gitignore (hunk 1 above)
Comment (friederbluemle) [reply .gitignore]: If you check the rest of the `.gitignore` file, you'll see that `node_modules/` is _already_ listed (in the default location for `.gitignore`). There is no need for this duplicate.

File: .gitignore (hunk 2 above)
Comment (friederbluemle) [reply .gitignore]: What do you mean by this? It's a comment that I removed. The file is more in line with the React Native default. Any specific requirements for this project?

File: example/.gitignore (hunk 3 above)
Comment (friederbluemle) [reply example/.gitignore]: With `yarn-lock` you mean `yarn.lock`? `yarn.lock` records a _known_ dependency tree for this example project (exact dependency versions that the project is guaranteed to work with). Please check this article for more info:
https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/

**Comment by friederbluemle** [comment]
Rebased onto latest `master` branch.

@Inbal-Tish - Could you please have another look at this?

**Pull request closed without merging**